metrics-scraper/monitoring.go

```diff
@@ -91,66 +125,84 @@ func run(
 		case <-ticker.C:
 			if err := pollData(ctx, c, config.Moniker, networkID); err != nil {
 				log.Printf("error polling data for %v: %v", config.Moniker, err)
+			} else {
+				log.Printf("data polled successfully for: %v", config.Moniker)
 			}
 		}
 	}
 }
 
 func pollData(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+	errs := []error{}
 	if err := sidecarsVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := nodeVersion(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
 	if err := latestBlockAndTimestamp(ctx, client, moniker, networkID); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
-	return nil
+	return errors.Join(errs...)
 }
 
 func latestBlockAndTimestamp(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
 	result := struct {
 		Number    string `json:"number"`
 		Timestamp string `json:"timestamp"`
 	}{}
+
 	err := client.CallContext(ctx, &result, ethGetBlockByNumberEndpoint, "latest", false)
 	if err != nil {
 		mezodUpGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest block to 0 for error
+		mezoLatestBlockGauge.WithLabelValues(moniker, networkID).Set(0)
+		// set latest timestamp to 0 for error
+		mezoLatestTimestampGauge.WithLabelValues(moniker, networkID).Set(0)
 		return fmt.Errorf("couldn't call %v for %v: %v", ethGetBlockByNumberEndpoint, moniker, err)
 	}
 
 	mezodUpGauge.WithLabelValues(moniker, networkID).Set(1)
```

Comment:
The fact that `mezoUpGauge` is managed inside `latestBlockAndTimestamp` is unfortunate. Let's move it up to `run`. If `pollData` was ok, up = 1. Otherwise, up = 0.

Reply:
addressed in: d1755782d3115b584b67e6fb763dc3d62386ef81